src/fromPairsMap.ts

```diff
@@ -0,0 +1,120 @@
+import _ from 'lodash'
+
+export default fromPairsMap
+
+/**
+ * Applies `fromPairs` to the result of mapping given `iteratee` to given
+ * collection.
+ *
+ * @param collection Collection to iterate over
+ * @param iteratee Function invoked per iteration
+ * @returns New object
+ */
+export function fromPairsMap<T, PropertyName, TResult> (
+  collection: T[] | null | undefined,
```

Comment:
can't this be defined also on a map?

Reply:
Yes but it's already covered in the Lodash module declaration.

You can't have multiple functions with the same name in Javascript. However, it's possible to have mutliple functions with the same name but different signature (parameters types) in TypeScript.

To achieve this TypeScript forces you to have only one implementation for that function but allows you to expose several signatures.

So when you see a 5 different possible `_.map` functions you are seeing 5 signatures for the very same implementation.

Here if we allow using Dictionaries as well as Arrays then user would be able to use types for which the underlying functions are not defined, for instance, an `ArrayIterator` used to map an `Dictionary`.

Lodash solves this by declaring completely different signatures for Arrays, Lists, Dictionaries and so on.

I've decided that, as users are going to use the signatures exposed in Lodash object, that's offering proper signatures there is goo enough. It's true that if you import the underlying method directly it won't allow you to use all the combinations allowed when using the mixin. However, that's not really a problem since the implementation is not designed to be used outside of a Lodash mixin.